### ISSUE TITLE
Enhancement suggestion: Read URLs from QR Codes

### ISSUE THREAD

**Issue by 3edc** [bitfireAT/davdroid#415]
Hi all, 
thanks for this wonderfull and very usefull app.

A small suggestion: Some URLs for CALDAV are quiet lengthy, e.g. when using OWCLOUD on a own hosted system at home.

As an alternative to entering CALDAV/CARDDAV URLs manually using the on-screen keyboard, reading and scanning a QR-image would be very helpfull. The QR may also contain the user credentials (username/password)

This would make configuration a breeze esp. for people that are not that technically affiliated. 

Best regards
Gerry

**Comment by LudovicRousseau** [comment]
Do you know a CALDAV/CARDDAV server that allows to provide a QR-code for the configuration URL?

Another idea is to transfer the configuration URL on the phone by email or something similar. And then copy/paste the URL from the email client to davdroid. Of course in this case do not include the password.

**Comment by devvv4ever** [comment]
DAVdroid supports resource detection/service discovery via .well-known and SRV/TXT. By configuring your server using one of these methods you'll be able to only enter your domain/hostname and DAVdroid then finds the resources automatically. No need for long URLs... ;) davdroid.bitfire.at/configuration has more information about it!

**Comment by phy25** [comment]
QR support is too heavy to add. I think it is more flexible if URLs like carddav|caldav|davdroid://username@example.com/dav/ can be handled to directly open the account setup page with information filled. Thus you can start the account creation by clicking a link on the webpage.

**Comment by untitaker** [comment]
There is a semi-standard for this, [``webcal://``-URLs](https://en.wikipedia.org/wiki/Webcal). Take a look at http://dmfs.org/wiki/index.php?title=ICloud for how CalDAV-Sync uses then.

**Comment by rfc2822** [comment]
Isn't WebCAL for non-CalDAV-calendar files? I.e. simple .ics files published on a HTTP server?

**Comment by untitaker** [comment]
Ahh, indeed the wiki article I linked doesn't use webcal URLs. Still, it'd be nice if DAVDroid's and @dmfs' apps would be compatible in that regard.

**Comment by MagicFab** [comment]
Several Firefox extensions generate a QRCode from whatever is in the address bar or from the clipboard. This would indeed save quite some time!

**Comment by IzzySoft** [comment]
> QR support is too heavy to add

Not necessarily. DAVDroid would only need to register for those "URL types", then any QR Code reader should list it in its "open-with" dialog.

> Several Firefox extensions generate a QRCode from whatever is in the address bar or from the clipboard.

Or even on right-clicking a link ("Create Barcode from Link" entry in the context menu with the "[Mobile Barcoder](https://addons.mozilla.org/de/firefox/addon/mobile-barcoder/)" extension).

> This would indeed save quite some time!

Definitely! Especially considering we still have to remove/re-create our accounts whenever a calendar/addressbook/… was added ;)

OTOH, one could simply scan those barcodes with a QR reader, and copy the URL to the clipboard to then paste it to DAVDroid. Still easier than typing.
